server/src/repositories/asset.repository.ts

```diff
@@ -219,7 +219,11 @@ export class AssetRepository implements IAssetRepository {
   getAll(pagination: PaginationOptions, options: AssetSearchOptions = {}): Paginated<AssetEntity> {
     let builder = this.repository.createQueryBuilder('asset').leftJoinAndSelect('asset.files', 'files');
     builder = searchAssetBuilder(builder, options);
-    builder.orderBy('asset.createdAt', options.orderDirection ?? 'ASC');
+    if (options.orderDirection === 'RAND') {
+      builder.addOrderBy('RANDOM()');
```

Comment:
I think this causes a full table scan unless I'm mistaken.

Reply:
It does, yes. The "official" fast way to do it is to use `TABLESAMPLE`, but it doesn't handle filters well since they apply *after* it fetches a row (or rows).

I think you have to get crafty to make it fast and work well with filters, like taking advantage of the id or checksum column's inherent randomization and comparing a random uuid or hash against the table.